src/logger/LoggerFactory.cpp

```diff
@@ -51,7 +51,7 @@ namespace mariadb
   bool LoggerFactory::initLoggersIfNeeded()
   {
     if (!NO_LOGGER) {
-      NO_LOGGER.reset(new NoLogger());
+      NO_LOGGER = Shared::Logger(new NoLogger());
```

Comment:
In that PR I asked "Is there real need for this change? I don't think so" Now I guess the idea is to initialize in case it's not initialized. Like calling method of not initialized object is not a good idea. But it's not initialization and one line before we test that object as boolean and that involves method calling.
In 1.1 this dance around NO_LOGGER has been removed and I should do the same in 1.0 probably. But as for your  PR  - I guess there is probably no need to change this line